src/SapAct/Extensions/JsonElementExtensions.cs

```diff
@@ -6,9 +6,7 @@ public static class JsonElementExtensions
 
 	public static List<ColumnDefinition> GenerateColumnList(this JsonElement payload, TargetStorageEnum targetStorage)
```

Comment:
Just as a thought, while we're touching this method, would it be worth trying to simplify this down with a bit of LINQ?

e.g.
```
var propDefinitions = payload.EnumerateObject()
  .SelectMany(p => p.Name == "data" ? p.Value.EnumerateObject() : [p])
  .Select(p => p.Name)
  .Distinct()
  .Select(name => new ColumnDefinition { Name = name, Type = "string" });

if (targetStorage is TargetStorageEnum.LogAnalytics)
{
  return propDefinitions
    .Append(new ColumnDefinition { Name = "TimeGenerated", Type = "datetime" })
    .ToList();
}
else
{
  return propDefinitions.ToList();
}
```